src/crypto.rs

```diff
@@ -16,15 +16,69 @@ pub trait Encode<I, O> {
     fn encode(&self, input: I) -> Self::ReturnType<O>;
 }
 
+pub trait Decode<I, O> {
+    type ReturnType<'b, T>
+    where
+        Self: 'b;
+    fn decode(&self, input: O) -> Self::ReturnType<'_, I>;
+}
+
+pub trait FromEncoded: Sized {
+    fn from_encoded(input: String) -> Option<Self>;
+}
+
+impl FromEncoded for masking::Secret<String> {
+    fn from_encoded(input: String) -> Option<Self> {
+        Some(input.into())
+    }
+}
+
+impl FromEncoded for Vec<u8> {
+    fn from_encoded(input: String) -> Option<Self> {
+        hex::decode(input).ok()
+    }
+}
+
 pub mod aes;
+#[cfg(feature = "aws-kms")]
```

Comment:
It might be a good idea to name these features using a common-prefix i.e.

`kms-aws`
`kms-hashicorp-vault` 

to give an indication that they are related.